miden-lib/tests/test_miden_wallet.rs

```diff
@@ -0,0 +1,312 @@
+pub mod common;
+
+use assembly::{
+    ast::{ModuleAst, ProgramAst},
+    Assembler,
+};
+use common::{data::prepare_word, NodeIndex};
+use crypto::{Felt, StarkField, Word, ONE, ZERO};
+use miden_lib::{MidenLib, SatKernel};
+use miden_stdlib::StdLibrary;
+
+use miden_objects::{
+    assets::{Asset, FungibleAsset},
+    builder::DEFAULT_ACCOUNT_CODE,
+    mock::{
+        mock_account_storage, mock_inputs_with_existing, AssetPreservationStatus, MockAccountType,
+        ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN, ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN,
+        ACCOUNT_ID_SENDER,
+    },
+    notes::{Note, NoteOrigin, NoteScript},
+    Account, AccountCode, AccountId, AccountVault, BlockHeader, ChainMmr,
+};
+
+use miden_tx::{data::DataStore, TransactionExecutor};
+
+#[derive(Clone)]
+pub enum DataStoreError {
+    AccountNotFound(AccountId),
+    NoteNotFound(u32, NodeIndex),
+}
+
+#[derive(Clone)]
+
+pub struct MockDataStore {
+    pub account: Account,
+    pub block_header: BlockHeader,
+    pub block_chain: ChainMmr,
+    pub notes: Vec<Note>,
+}
```

Comment:
It seems like this code is identical to the one in `miden-tx`. Maybe a more natural place for these tests would actually be in `moden-tx` crate? This way, we won't need to duplicate code and should avoid the circular dependency issue.

Reply:
Yes, that's right. However, to me, the most natural place to test the basic MASM procedures would be where they are defined. 

I hope we can get rid of this code when we have @hackaugusto mock data framework. So I would suggest to still test the wallet where it is defined, in `miden-lib`, but get rid of the code duplication once we have a better testing framework. Does that make sense?

Reply:
I can see it both ways actually. Yes, the code is defined here, but we are running a full "integration test" of executing a transaction which is defined in `miden-lib`. So, I think both are fine.

We can leave it as is for now, but I think we'll need to come back to this when we refactor the dependencies as mentioned in one other comment.